dvc/tree/s3.py

```diff
@@ -75,24 +73,52 @@ def s3(self):
 
         session = boto3.session.Session(**session_opts)
 
-        return session.client(
+        return session.resource(
```

Comment:
`resource` is a higher level API, so we don't have to manually parse-out HTTP codes.